src/app/application.cpp

```diff
@@ -688,74 +702,157 @@ int Application::exec(const QStringList &params)
-    });
-    connect(btSession, &BitTorrent::Session::downloadFromUrlFailed, this
-            , [this](const QString &url, const QString &reason)
-    {
-        m_desktopIntegration->showNotification(tr("URL download error")
-            , tr("Couldn't download file at URL '%1', reason: %2.").arg(url, reason));
-    });
+#ifndef DISABLE_WEBUI
+        m_webui = new WebUI(this);
+#ifdef DISABLE_GUI
+        if (m_webui->isErrored())
+            QCoreApplication::exit(1);
+        connect(m_webui, &WebUI::fatalError, this, []() { QCoreApplication::exit(1); });
+
+        const Preferences *pref = Preferences::instance();
+
+        const auto scheme = pref->isWebUiHttpsEnabled() ? u"https"_qs : u"http"_qs;
+        const auto url = u"%1://localhost:%2\n"_qs.arg(scheme, QString::number(pref->getWebUiPort()));
+        const QString mesg = u"\n******** %1 ********\n"_qs.arg(tr("Information"))
+                + tr("To control qBittorrent, access the WebUI at: %1").arg(url);
+        printf("%s\n", qUtf8Printable(mesg));
 
-    m_window = new MainWindow(this);
+        if (pref->getWebUIPassword() == QByteArrayLiteral("ARQ77eY1NUZaQsuDHbIMCA==:0WMRkYTUWVT9wVvdDtHAjU9b3b7uB8NR1Gur2hmQCvCDpm39Q+PsJRJPaCU51dEiz+dTzh8qbPsL8WkFljQYFQ=="))
+        {
+            const QString warning = tr("The Web UI administrator username is: %1").arg(pref->getWebUiUsername()) + u'\n'
+                    + tr("The Web UI administrator password has not been changed from the default: %1").arg(u"adminadmin"_qs) + u'\n'
+                    + tr("This is a security risk, please change your password in program preferences.") + u'\n';
+            printf("%s", qUtf8Printable(warning));
+        }
 #endif // DISABLE_GUI
+#endif // DISABLE_WEBUI
+
+        m_isProcessingParamsAllowed = true;
+        for (const AddTorrentParams &params : m_paramsQueue)
+            processParams(params);
+        m_paramsQueue.clear();
+    });
 
     if (!params.isEmpty())
         m_paramsQueue.append(parseParams(params));
 
     return BaseApplication::exec();
 }
+catch (const RuntimeError &err)
+{
+#ifdef DISABLE_GUI
+    fprintf(stderr, "%s", qPrintable(err.message()));
+#else
+    QMessageBox msgBox;
+    msgBox.setIcon(QMessageBox::Critical);
+    msgBox.setText(tr("Application failed to start."));
+    msgBox.setInformativeText(err.message());
+    msgBox.show(); // Need to be shown or to moveToCenter does not work
+    msgBox.move(Utils::Gui::screenCenter(&msgBox));
+    msgBox.exec();
+#endif
+    return EXIT_FAILURE;
+}
 
 bool Application::isRunning()
 {
     return !m_instanceManager->isFirstInstance();
 }
 
 #ifndef DISABLE_GUI
+void Application::createStartupProgressDialog()
+{
+    Q_ASSERT(m_desktopIntegration);
+#if (QT_VERSION < QT_VERSION_CHECK(6, 0, 0))
+    disconnect(m_desktopIntegration, &DesktopIntegration::activationRequested, this, &Application::createStartupProgressDialog);
+#endif
+
+    m_startupProgressDialog = new QProgressDialog;
```

Comment:
I would use another ctor https://doc.qt.io/qt-6/qprogressdialog.html#QProgressDialog-1 that can take more arguments at once.

Reply:
> I would use another ctor https://doc.qt.io/qt-6/qprogressdialog.html#QProgressDialog-1 that can take more arguments at once.

Ok.

Reply:
But I will prefer to add `WindowMinimizeButtonHint` flag separately. Otherwise, I would have to specify all the necessary flags in the constructor, which looks unnecessarily cumbersome.